integration/kotlinx-coroutines-jdk8/src/future/Future.kt

```diff
@@ -86,6 +82,28 @@ public fun <T> Deferred<T>.asCompletableFuture(): CompletableFuture<T> {
     return future
 }
 
+/**
+ * Converts this completable job to the instance of [CompletableFuture].
+ * The job is cancelled when the resulting future is cancelled or otherwise completed.
+ */
+public fun CompletableJob.asCompletableFuture(): CompletableFuture<Unit> {
```

Comment:
It has to be extension on `Job`, not `CompletableJob`, just like we have extension on `Deferred` and we don't need `CompletableDeferred`, since the value link is only one way, and only cancellation goes the other way around.